### PR TITLE
Add TRACE logging level for verbose payload logging across event writers

### LINKED ISSUE #59: Add trace logging with the payloads
## Background
Debug level sometimes is not enough.

## Feature
A tracing level logging needs to be added which will include the payload the is service trying to send to each of the services (kafka, eventbridge, postgres)

### PR DESCRIPTION
**Pull request by oto-macenauer-absa**
This pull request introduces a new custom TRACE logging level (below DEBUG) for highly verbose payload logging, and integrates it into the EventBridge, Kafka, and Postgres writer modules. It also includes comprehensive tests to ensure TRACE-level logging works as intended across all three writers. The main themes are the addition of the TRACE level, its integration into the writers, and the new tests for this functionality.

**Custom TRACE logging level:**

* Added a new `TRACE_LEVEL` (level 5) and a corresponding `Logger.trace()` method in `logging_levels.py`, ensuring idempotent registration and compatibility with the standard logging module.
* Updated module imports in `event_gate_lambda.py`, `writer_eventbridge.py`, `writer_kafka.py`, and `writer_postgres.py` to ensure the TRACE level is registered before use. [[1]](diffhunk://#diff-c4dd7947b9e048d7d7e0b6f173972ce2980c114f0175b06f301f24a578928df1R32-R37) [[2]](diffhunk://#diff-e6c2ff45b654b1e9b95f8280e35ec71e4f59fbdbf5297ad099bafba4ae7c4a9dR13-R16) [[3]](diffhunk://#diff-4b121ee43b4a4565922497c29c7f8bc0077a20d7b03c6ca8542fcb1da7185a0aR13-R15) [[4]](diffhunk://#diff-853878d80bdbb8c779f4ca51d8e96a99ced6dd5019f757534fbc7b4150505a8aR18-R20)

**Integration of TRACE logging in writers:**

* In `writer_eventbridge.py`, `writer_kafka.py`, and `writer_postgres.py`, added TRACE-level payload logging before sending messages, including defensive handling for unserializable payloads. [[1]](diffhunk://#diff-e6c2ff45b654b1e9b95f8280e35ec71e4f59fbdbf5297ad099bafba4ae7c4a9dR59-R67) [[2]](diffhunk://#diff-4b121ee43b4a4565922497c29c7f8bc0077a20d7b03c6ca8542fcb1da7185a0aR81-R89) [[3]](diffhunk://#diff-853878d80bdbb8c779f4ca51d8e96a99ced6dd5019f757534fbc7b4150505a8aR246-R254)

**Testing:**

* Added `test_trace_logging.py` to verify that TRACE-level logging is triggered and correctly logs payloads for EventBridge, Kafka, and Postgres writers, using mocks and monkeypatching as needed.

**Release notes**
- Add TRACE logging level for verbose payload logging across event writers

**Related**
- Closes #59 

## Summary by CodeRabbit

- **New Features**
  - Added a TRACE logging level with logger.trace(...) support and exported TRACE_LEVEL.
  - Enabled optional TRACE-level payload logging for EventBridge, Kafka, and Postgres writers via a unified helper.
  - Added safe, PII-redacting, size-bounded serialization for trace payloads (preserves UTF-8, truncates safely).

- **Tests**
  - Added comprehensive tests for TRACE payload logging and safe serialization across backends and edge cases.

### CODE REVIEW

File: src/writer_postgres.py [hunk 1]
```diff
@@ -271,6 +274,15 @@ def write(topic_name: str, message: Dict[str, Any]) -> Tuple[bool, Optional[str]
             _logger.debug("psycopg2 not available - skipping actual Postgres write")
             return True, None
 
+        # TRACE-level payload logging (only when we intend to write)
+        if _logger.isEnabledFor(TRACE_LEVEL):
+            try:
+                _logger.trace(  # type: ignore[attr-defined]
+                    "Postgres payload topic=%s payload=%s", topic_name, json.dumps(message, separators=(",", ":"))
+                )
+            except Exception:  # pragma: no cover - defensive
```
Comment (tmikula-dev) [inline src/writer_postgres.py]:
```suggestion
            except (TypeError, ValueError):  # pragma: no cover - defensive serialization guard
```

File: src/writer_postgres.py [hunk 2]
```diff
@@ -271,6 +274,15 @@ def write(topic_name: str, message: Dict[str, Any]) -> Tuple[bool, Optional[str]
...
             return True, None
 
+        # TRACE-level payload logging (only when we intend to write)
+        if _logger.isEnabledFor(TRACE_LEVEL):
+            try:
+                _logger.trace(  # type: ignore[attr-defined]
+                    "Postgres payload topic=%s payload=%s", topic_name, json.dumps(message, separators=(",", ":"))
+                )
+            except Exception:  # pragma: no cover - defensive
+                _logger.trace("Postgres payload topic=%s <unserializable>", topic_name)  # type: ignore[attr-defined]
```
Comment (tmikula-dev) [inline src/writer_postgres.py]: Please why the logger in this script starts with `_`?

File: src/writer_kafka.py [hunk 3]
```diff
@@ -91,6 +94,15 @@ def write(topic_name: str, message: Dict[str, Any]) -> Tuple[bool, Optional[str]
         logger.debug("Kafka producer not initialized - skipping")
         return True, None
 
+    # TRACE-level payload logging prior to produce
+    if logger.isEnabledFor(TRACE_LEVEL):
+        try:
+            logger.trace(  # type: ignore[attr-defined]
+                "Kafka payload topic=%s payload=%s", topic_name, json.dumps(message, separators=(",", ":"))
+            )
+        except Exception:  # pragma: no cover - defensive
```
[the remark this answers is not in the document]
Comment (oto-macenauer-absa) [reply src/writer_kafka.py]: Fixed

File: src/writer_eventbridge.py [hunk 4]
```diff
@@ -68,6 +72,15 @@ def write(topic_name: str, message: Dict[str, Any]) -> Tuple[bool, Optional[str]
...
 
+    # TRACE-level payload logging
+    if logger.isEnabledFor(TRACE_LEVEL):
+        try:
+            logger.trace(  # type: ignore[attr-defined]
+                "EventBridge payload topic=%s payload=%s", topic_name, json.dumps(message, separators=(",", ":"))
+            )
+        except Exception:  # pragma: no cover - defensive serialization guard
+            logger.trace("EventBridge payload topic=%s <unserializable>", topic_name)  # type: ignore[attr-defined]
+
```
[the remark this answers is not in the document]
Comment (oto-macenauer-absa) [reply src/writer_eventbridge.py]: Added trace helper which handles PII and truncating. Please check.

**Review by tmikula-dev — APPROVED** [review]
Great, thanks for the changes. If you have time, I have one more recommendation to create some source structure. The project is getting bigger and we should probably create a file for writers/ and logging/ or utils/ keeping the event_gate_lambda.py script as main one.
